pkg/git/url.go

```diff
@@ -41,3 +42,15 @@ func (r Remote) Equivalent(u string) bool {
 	}
 	return lu.Host == ru.Host && trimPath(lu.Path) == trimPath(ru.Path)
 }
+
+// HasValidHostname checks the hostname in repo url is valid format
+func (r Remote) HasValidHostname() (string, bool) {
+	arr := strings.Split(r.URL, "@")
+	hostname := strings.Split(arr[1], ":")[0]
+
+	re, _ := regexp.Compile(`^[a-zA-Z0-9][a-zA-Z0-9-]{1,61}[a-zA-Z0-9]\.[a-zA-Z]{2,}$`)
```

Comment:
```golang
	re, _ := regexp.Compile(`^((git|ssh|http(s)?)|(git@[\w\.]+))(:(//)?)([\w\.@\:/\-~]+)((\.git)?)(/)?`)
	if re.MatchString(r.URL) {
	...
```

[Based on this](https://stackoverflow.com/a/22312124), while taking into account that for e.g. GitHub an URL ending with `.git` isn't a strict requirement.

Reply:
Actually, this does not validate the hostname. It validates the invalid hostnames such as `githubcom`.

And my solution only purposed on validating hostname format, so it should get hostname from the repo URL. It does not check the host domain really exists.

Reply:
>  It validates the invalid hostnames such as `githubcom`.

`githubcom` can be a valid hostname when for example a custom entry in `/etc/hosts` exists, and in addition when people are using a custom Git host, IP addresses can be valid entries too. The current regex does not take those into account, and I actually think it will be really hard to reliably validate all of them, as they can technically be any value due to `/etc/hosts` entries.

Another option to just validate the presence of a hostname would be to use [`url.Parse`](https://golang.org/pkg/net/url/#Parse) on the `r.URL`, returning `false` if this returns a parsing error, or further validating the presence of [`URL.Hostname`](https://golang.org/pkg/net/url/#URL.Hostname) if the parsing succeeds.

Reply:
The `url.Parse` can not parse some valid ssh URLs such as `git@github.coom/fluxcd/flux`.
So because of those who use custom entities which are not valid host domain format for repo hosting, I think we have to rely on the fluxctl sync result for better logging.

I will change the commit and PR title positively.

Reply:
Hi @hiddeco ,
Can you please review the new commit?

It parses the fluxctl sync logs and shows error msg based on that.
Here is the error message.
`Error: can't connect to git repo with url ssh://git@github/sysdevguru/flux-get-started`